local-cli/runAndroid/runAndroid.js

```diff
@@ -12,91 +12,64 @@ const chalk = require('chalk');
 const child_process = require('child_process');
 const fs = require('fs');
 const path = require('path');
-const parseCommandLine = require('../util/parseCommandLine');
 const isPackagerRunning = require('../util/isPackagerRunning');
 const Promise = require('promise');
 const adb = require('./adb');
 
+// Verifies this is an Android project
+function checkAndroid(root) {
+  return fs.existsSync(path.join(root, 'android/gradlew'));
+}
+
 /**
  * Starts the app on a connected Android emulator or device.
  */
-function runAndroid(argv, config) {
-  return new Promise((resolve, reject) => {
-    _runAndroid(argv, config, resolve, reject);
-  });
-}
-
-function _runAndroid(argv, config, resolve, reject) {
-  const args = parseCommandLine([{
-    command: 'install-debug',
-    type: 'string',
-    required: false,
-  }, {
-    command: 'root',
-    type: 'string',
-    description: 'Override the root directory for the android build (which contains the android directory)',
-  }, {
-    command: 'flavor',
-    type: 'string',
-    required: false,
-  }, {
-    command: 'variant',
-    type: 'string',
-    required: false,
-  }], argv);
-
-  args.root = args.root || '';
-
-  if (!checkAndroid(args)) {
+function runAndroid(argv, config, args) {
+  if (!checkAndroid(args.root)) {
     console.log(chalk.red('Android project not found. Maybe run react-native android first?'));
```

Comment:
@mkonicek in next PR, what I would like to see is to move these all chalk.red calls (errors) to be just Promise.reject / throw new Error so that we have a centralised error reported in the `cliEntry.js` file. That would also stay consistent with other commands (like runIOS) where we throw errors instead of logging & swallowing them. Might be also a bit easier to work with in case one is debugging and doesn't know where the error is coming from.